README.md

```diff
@@ -2,107 +2,112 @@
+The design of this library was originally inspired by OpenZeppelin's [access control library](https://github.com/OpenZeppelin/cairo-contracts), as well as Python's [flags](https://docs.python.org/3/library/enum.html) and Vyper's [enums](https://docs.vyperlang.org/en/stable/types.html#enums).
 
 ## Overview
 
-This library uses felt values in the form of 2<sup>n</sup>, where `n` is in the range `0 <= n <= 251`, to represent user-defined roles as members. 
+This library uses `u128` values in the form of 2<sup>n</sup>, where `n` is in the range `0 <= n < 128`, to represent user-defined roles as members. 
 
-Roles should be defined in a separate Cairo contract as its own namespace. For example:
+We recommend users to define the roles in a separate Cairo file. For example:
 
 ```cairo
-namespace Roles {
-    const MANAGER = 2 ** 0;
-    const STAFF = 2 ** 1;
-    const USER = 2 ** 2;
+mod user_roles {
+    const MANAGER: u128 = 1;
+    const STAFF: u128 = 2;
+    const USER: u128 = 4;
 }
 ```
 
-Multiple roles can be represented as a single value by performing bitwise AND. Using the above example, an address can be assigned both the `MANAGER` and `STAFF` roles using a single value of 3 (equivalent to `Roles.MANAGER | Roles.STAFF` or `2 ** 0 + 2 ** 1`).
+Multiple roles can be represented as a single value by performing bitwise AND. Using the above example, an address can be assigned both the `MANAGER` and `STAFF` roles using a single value of 3 (equivalent to `user_roles::MANAGER | user_roles::STAFF` or `2 ** 0 + 2 ** 1`).
 
 Similarly, multiple roles can be granted, revoked or checked for in a single transaction using bitwise operations:
 - granting role(s) is a bitwise AND operation of the currently assigned value and the value of the new role(s);
 - revoking role(s) is a bitwise AND operation of the currently assigned value and the complement (bitwise NOT) of the value of the role(s) to be revoked; and
 - checking for membership is a bitwise OR operation of the currently assigned value and the value of the role(s) being checked for.
 
-Note that functions which rely on this access control library will require the `bitwise_ptr` implicit argument and `BitwiseBuiltin`.
-
-
 ## Usage
 
-To use this library in a Cairo contract:
-1. Include a copy of `accesscontrol_library.cairo` in your project, and import the library into the Cairo contract.
-2. Define the available roles as constants in a namespace in a separate Cairo contract, and import this namespace into the Cairo contract.
+To use this library, add the repository as a dependency in your `Scarb.toml`:
 
-For example, assuming you have a `contracts/` folder with `accesscontrol_library.cairo` and `roles.cairo`, and you want to import both into a Cairo file within the same folder:
+```
+[dependencies]
+wadray = { git = "https://github.com/lindy-labs/cairo-wadray.git" }
```

Comment:
![](https://media.giphy.com/media/51Uiuy5QBZNkoF3b2Z/giphy-downsized-large.gif)

Reply:
Shall we also rename the repo to `accesscontrol`?

Reply:
Yes. Already pinged @0xBRM about it because I don't have the rights 😢 But go ahead and do it if you do.

Reply:
Yes, I can change it on my end. Slightly pedantic, but should we go with `access_control` now?

Reply:
Yes, 100%. That's better.